pkg/cloud/services/compute/instance.go

```diff
@@ -562,38 +561,22 @@ func applyServerGroupID(opts servers.CreateOptsBuilder, serverGroupID string) se
 	return opts
 }
 
-// Helper function for getting image id from name.
-func (s *Service) getImageIDFromName(imageName string) (string, error) {
-	var opts images.ListOpts
-
-	opts.Name = imageName
-
-	allImages, err := s.getImageClient().ListImages(opts)
+// Helper function for getting image ID from name or ID.
+func (s *Service) getImageID(image infrav1.ImageFilter) (string, error) {
+	allImages, err := s.getImageClient().ListImages(image.ToListOpt())
```

Comment:
should we follow https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1779/files#diff-2669f4e6453b138e99121b1203a91f87fa3bc88468c8873c1912c0492566075fR28 ?

Reply:
good idea, i'm doing this.